Userland/Libraries/LibJS/Runtime/RegExpObject.h

```diff
@@ -49,13 +49,16 @@ class RegExpObject : public Object {
     String const& flags() const { return m_flags; }
     Regex<ECMA262> const& regex() { return *m_regex; }
     Regex<ECMA262> const& regex() const { return *m_regex; }
+    bool has_legacy_features() const { return m_legacy_features; }
+    void set_legacy_features(bool flag) { m_legacy_features = flag; }
 
 private:
     RegExpObject(Object& prototype);
     RegExpObject(Regex<ECMA262> regex, String pattern, String flags, Object& prototype);
 
     String m_pattern;
     String m_flags;
+    bool m_legacy_features { true };
```

Comment:
```suggestion
    bool m_has_legacy_features { true };
```

Also, this doesn't sound like a sane default.